substrate/frame/revive/src/storage.rs

```diff
@@ -238,8 +242,9 @@ impl<T: Config> ContractInfo<T> {
 	/// The base deposit is updated when the `code_hash` of the contract changes, as it depends on
 	/// the deposit paid to upload the contract's code.
 	pub fn update_base_deposit(&mut self, code_info: &CodeInfo<T>) -> BalanceOf<T> {
+		// The 2 items added are code info and immutable data
 		let info_deposit =
-			Diff { bytes_added: self.encoded_size() as u32, items_added: 1, ..Default::default() }
+			Diff { bytes_added: self.encoded_size() as u32, items_added: 2, ..Default::default() }
```

Comment:
While the new approach is correct it has a slight wrinkle: We charge for the storage item even if no immutable data is used by the contract. I think this change here should be reverted and the charging of the item should be done in `set_immutable_bytes`, too: Charge for the item iff `immutable_bytes > 0 `.

Reply:
Yeah I agree that's not proper. I think this works if we only allow calling `set_immutable_bytes` once (otherwise  need to track whether the item is already charged). But this restriction is fine anyways.

Reply:
We don't actually need to track anything if we forbid setting empty immutable data with a length of 0, since this implies it. I implemented it like this. Let me know if you rather want to track it explicitly.

Reply:
If the constraint of only calling it once is fine for revive then this is fine.

Reply:
I don't see a use case for calling it multiple times. Also, the matter-labs implementation returns the immutable data from the constructor (instead of the bytecode), so it should be fine anyways :)